src/tools/synthetic_node.rs

```diff
@@ -430,6 +430,10 @@ impl InnerNode {
     async fn send_direct_bytes(&self, target: SocketAddr, data: Vec<u8>) -> io::Result<()> {
         self.node.send_direct_message(target, data.into()).await
     }
+
+    fn connected_peers(&self) -> Vec<SocketAddr> {
```

Comment:
We may want this to be public? 
```suggestion
    /// Returns the list of active connections for this node. Should be prefered over [`known_peers`] when querying active connections.
    pub fn connected_peers(&self) -> Vec<SocketAddr> {
```